### ISSUE TITLE
Python Segmentation fault (core dumped) on cuda11.0-runtime when built on cuda11.0-devel

### ISSUE THREAD

**Issue by polar99** [microsoft/onnxruntime#6339]
I have built onnxruntime from master with CUDA11.0 cudnn8 (cuda-devel) . I found that onnxruntime-gpu works well on cuda-devel environment, but it failed on cuda-runtime when running onnxruntime.InferenceSession() with yolov5 model.
The exp_yolov5.py [code](https://github.com/polar99/onnxruntime-gpu-test/blob/main/exp_yolov5.py)
```
import onnxruntime

onnx_path = './yolov5s.onnx'
onnxruntime.InferenceSession(onnx_path, providers=['CUDAExecutionProvider'])
```
The onnxfile [yolov5s_export_false.onnx](https://drive.google.com/file/d/1TxmRqIFcR474vCzaDJhjIS-gBCJPwtRk/view?usp=sharing) (Details of this yolov5s.onnx are in section 3 ) 

### 1. Build based environment in docker containers

**cuda11.0-devel** environment: docker pull nvidia/cuda:11.0-cudnn8-devel-ubuntu18.04
**cuda11.0-runtime** environment: docker pull nvidia/cuda:11.0-cudnn8-runtime-ubuntu18.04

I have install same software both in **cuda11.0-devel** and **cuda11.0-runtime**.
(1) based software
`apt install build-essential zlib1g-dev libncurses5-dev libgdbm-dev libnss3-dev libssl-dev libreadline-dev libffi-dev wget python3-setuptools python3-numpy python3-wheel python python3-pip python3-pytest python3-distutils git curl libgl1-mesa-glx libglib2.0-dev language-pack-en -y`

```
locale-gen en_US.UTF-8
update-locale LANG=en_US.UTF-8
```

(2) Install Python3.8.7 from [here](https://www.python.org/ftp/python/3.8.7/Python-3.8.7.tgz)
(3) Install Cmake3.19.2 from [official cmake website](https://cmake.org/download/)

### 2. Compile onnxruntime-gpu

(1)**cuda-devel** environment: docker pull nvidia/cuda:11.0-cudnn8-devel-ubuntu18.04
The onnxruntime is built from master on this environment.
When running
`python exp_yolov5.py`
onnxruntime works well.

(2)**cuda-runtime** environment: docker pull nvidia/cuda:11.0-cudnn8-runtime-ubuntu18.04
The onnxruntime is built from master on cuda-devel environment.
When running
`python exp_yolov5.py`
onnxruntime will cause fatal error.
Segmentation fault (core dumped)

The following is my building cmd:
./build.sh --build_shared --parallel --use_cuda --enable_pybind --build_wheel --config Debug--cudnn_home /usr/lib/x86_64-linux-gnu/ --cuda_home /usr/local/cuda --cmake_extra_defines PYTHON_LIBRARIES=/usr/lib/x86_64-linux-gnu/libpython3.8.so.1.0 PYTHON_EXECUTABLE=/usr/local/bin/python3.8 PYTHON_INCLUDE_DIRS=/usr/local/include/python3.8

Finally, I got a python wheel then installed onnxruntime_gpu by pip.
onnxruntime_gpu-1.6.0-cp38-cp38-linux_x86_64.whl, download here [Google Drive](https://drive.google.com/file/d/18GJ5GbC3KrMUz4KaaSUuPQ5_U77Rp2Ix/view?usp=sharing)

**GDB debug info:** (on cuda-runtime)
![image](https://user-images.githubusercontent.com/23132594/104419791-ee45f080-55b3-11eb-8638-3a63d81bbac0.png)

### 3. About exporting yolov5s.onnx
(1) Download yolov5s.pt from [official release](https://github.com/ultralytics/yolov5/releases/tag/v4.0) 
(2) Modify [export.py ](https://github.com/ultralytics/yolov5/blob/v4.0/models/export.py)
In export.py [Line 53](https://github.com/ultralytics/yolov5/blob/69be8e738f45e7908c1270eedd350f98c0c7bfa4/models/export.py#L53)  
  `model.model[-1].export = True  # set Detect() layer export=True`  
As I want to output post-processed boxes from yolov5 network, I have set 
` model.model[-1].export = False`
This bool value will change the output onnx file.

More details in [yolo.py](https://github.com/ultralytics/yolov5/blob/v4.0/models/yolo.py)  
and the [line](https://github.com/ultralytics/yolov5/blob/69be8e738f45e7908c1270eedd350f98c0c7bfa4/models/yolo.py#L48)
```
            if not self.training:  # inference
                if self.grid[i].shape[2:4] != x[i].shape[2:4]:
                    self.grid[i] = self._make_grid(nx, ny).to(x[i].device)

                y = x[i].sigmoid()
                y[..., 0:2] = (y[..., 0:2] * 2. - 0.5 + self.grid[i].to(x[i].device)) * self.stride[i]  # xy
                y[..., 2:4] = (y[..., 2:4] * 2) ** 2 * self.anchor_grid[i]  # wh
                z.append(y.view(bs, -1, self.no))
```
In conclusion, exported onnx file ([yolov5s_export_false.onnx](https://drive.google.com/file/d/1TxmRqIFcR474vCzaDJhjIS-gBCJPwtRk/view?usp=sharing)) will cause Segmentation fault (core dumped) when model.model[-1].export = False on cuda11.0-runtime, but running normally on cuda11.0-devel.

If applying model.model[-1].export = True, exported onnx file ([yolov5s_export_true.onnx](https://drive.google.com/file/d/1YwcLrQx-FpSjI3FftNokNWxLFh8BWdTH/view?usp=sharing)) will running normally both on cuda11.0-devel and cuda11.0-runtime.

This problem is very strange. I don't know if the build environment is set up correctly. That's probably why.

**Comment by snnn** [comment]
This issue has been automatically closed as 'not planned' because it has been marked as 'stale' for more than 30 days without activity. If you believe this is still an issue, please feel free to reopen it.
